### PR TITLE
Update 5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar

### PR DESCRIPTION
**Pull request by nerdsanddragons**
Fixed bug in -loadmap feature

This file should be up to date

### CODE REVIEW

**Comment by nerdsanddragons** [comment]
Delete my last pull request

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 1]
```diff
@@ -84,56 +85,45 @@ if args.last('viewload') or args.last('loadview'):
...
+    # We only care about the first result	
+    break	
+</drac2>	
+<drac2>
+# Pulling up viewlist to view all created -mapview presets 
+userPresets = load_json(get('mapPresets','{}'))
+# Only select the ones they are searching for, if they are searching
+viewlist = [f"""**{presets}** - `{str(over).replace('"',quote)}`""" for presets, over in userPresets.items() if args.last('search','').lower() in presets.lower()]
+# Split the list into groups of 20, to ensure it stays a reasonable size
+presetPagin = [viewlist[i:i+20] for i in range(0, len(viewlist), 20)]
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:110]: This adds the new viewlist to show all the saved -mapview presets you made

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 2]
```diff
@@ -84,56 +85,45 @@ if args.last('viewload') or args.last('loadview'):
...
-  if (args.last('mapload') or args.last('loadmap')).lower() in maps.lower():
-   # If we find it, add the map to the args
-   foundMapload = mapBG[maps]
-   if typeof(foundMapload) == 'str':
-    args = argparse(["-walls",  foundMapload] + "@@@")
-   elif typeof(foundMapload) == "SafeList":
-    getBG, getSize, getOptions = (foundMapload + ["", ""])[:3]
-    args = argparse(["-mapbg", getBG]   + (["-mapsize"  , getSize] if getSize else []) + (["-mapoptions", getOptions]  if getOptions else []) + "@@@")
-   break
+objectPagin = [objectlist[i:i+20] for i in range(0, len(objectlist), 20)]
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:118]: This is the new objectlist

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 3]
```diff
@@ -84,56 +85,45 @@ if args.last('viewload') or args.last('loadview'):
...
-# Pulling up viewlist to view all created -mapview presets 
-userPresets = load_json(get('mapPresets','{}'))
+# Pulling up tokenlist to view all created tokens 
+userTokens = load_json(get('mapTokens','{}'))
 # Only select the ones they are searching for, if they are searching
-viewlist = [f"""**{presets}** - `{str(over).replace('"',quote)}`""" for presets, over in userPresets.items() if args.last('search','').lower() in presets.lower()]
+tokenlist = [f"""**{tokens}** - `{str(over).replace('"',quote)}`""" for tokens, over in userTokens.items() if args.last('search','').lower() in tokens.lower()]
 # Split the list into groups of 20, to ensure it stays a reasonable size
-presetPagin = [viewlist[i:i+20] for i in range(0, len(viewlist), 20)]
+tokenPagin = [tokenlist[i:i+20] for i in range(0, len(tokenlist), 20)]
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:126]: This is the new tokenlist

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 4]
```diff
@@ -169,14 +159,17 @@ if c:
   mapsize = f"{mapX}x{mapY}" 
   mapbg=mapinfo.get('background')
   mapoptions=mapinfo.get('options')
-  walls=mapinfo.get('walls')
+  walls = mapinfo.get('walls')
+  walls = walls.split(', ') if walls else []
+  objects = mapinfo.get('objects')
+  objects = objects.split('/') if objects else []
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:165]: Added this so the new -walls allows you to add them one at a time as well as -object one at a time

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 5]
```diff
@@ -266,11 +259,48 @@ if c:
...
+   foundObject = "There is no object by that name"
+   argsObject = args.last("removeobject").lower().split(":")
+   for object in presetObjects:
+    if argsObject[1] in object.lower():
+     foundObject = presetObjects[object]
+     break
+   objectLocation = argsObject[0]
+   loadedObject=objectLocation+foundObject
+   objects.remove(loadedObject)
+   desc.append(f"Object `{object}:{loadedObject}` was removed")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:303]: Updated walls so you can add them one at a time and added the new feature objects

Add and remove  for both

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 6]
```diff
@@ -282,8 +312,7 @@ if c:
   if mapattach and mapattach.get_effect('map'):
    mapattach.remove_effect('map')
   # Format everything appropriately
-  setMapView = f"{mapviewlocation}:{mapviewsize}"
-  neweffect = f"""{f"Size: {mapsize}" if mapsize else ""}{f" ~ Background: {mapbg}" if mapbg else ""}{f" ~ Options: {mapoptions}" if mapoptions else ""}{f" ~ Walls: {walls}" if walls else""}{f" ~ View: {setMapView}" if setMapView.strip(':') else ''}""".strip(" ~")
+  neweffect = f"""{f"Size: {mapsize}" if mapsize else ""}{f" ~ Background: {mapbg}" if mapbg else ""}{f" ~ Options: {mapoptions}" if mapoptions else ""}{f' ~ Walls: {", ".join(walls)}' if walls else ""}{f' ~ Objects: {"/".join(objects)}' if objects else ""}{f" ~ View: {mapviewlocation}:{mapviewsize}" if mapviewsize and mapviewlocation else ""}""".strip(" ~")
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:315]: This adds the new walls and new object to the mapattach atack effect

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 7]
```diff
@@ -293,71 +322,120 @@ if c:
...
-<drac2>
 #This allows the user to save -mapview presets to their mapPresets uvar to be called using -loadview
-if args.last('newview') or args.last('deleteview'):
+if args.last('saveview') or args.last('deleteview'):
   #First option is to create the preset
   userPresets=load_json(get("mapPresets", "{}"))
-  if args.last('newview'):
-   for addNewview in args.get('newview'):
+  if args.last('saveview'):
+   for addNewview in args.get('saveview'):
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:330]: All this was changing the language to keep everything consistent

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 8]
```diff
@@ -293,71 +322,120 @@ if c:
...
+    if mapinfo.get('options')!= None:
+     mapBGState["mapoptions"] = mapinfo.get('options')
+    if mapinfo.get('walls')!= None:
+     mapBGState["walls"] = mapinfo.get('walls').split(', ')
+    if mapinfo.get('objects')!= None:
+     mapBGState["objects"] = mapinfo.get('objects').split('/')
+  userMaps.update(formedMap)
+  set_uvar("mapBackgrounds",dump_json(userMaps))
+  desc.append(f"Created the map `{mapName}`")
+ #Second option is to delete an already created map
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:417]: New savemap...saves everything that is saved in mapattach..So user just ahs to run !map -savemap (mapname)

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 9]
```diff
@@ -293,71 +322,120 @@ if c:
...
+   	mapsize = foundMap[1]
+   	mapoptions = foundMap[2]
+   	objects = []
+   	walls = []
+   desc.append(f"Loaded the map {maps}")
+   neweffect = f"""{f"Size: {mapsize}" if mapsize else ""}{f" ~ Background: {mapbg}" if mapbg else ""}{f" ~ Options: {mapoptions}" if mapoptions else ""}{f' ~ Walls: {", ".join(walls)}' if walls else ""}{f' ~ Objects: {"/".join(objects)}' if objects else ""}{f" ~ View: {mapviewlocation}:{mapviewsize}" if mapviewsize and mapviewlocation else ""}""".strip(" ~")
+   if mapattach:
+    mapattach.add_effect('map', f"""-attack "||{neweffect}" """) 
+   break
+</drac2>
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:393]: New loadmap to work with the new savemap..it accounts for the older way the maps were being saved and will still load them

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 10]
```diff
@@ -565,6 +643,19 @@ if c:
...
+ if args.last('loadtoken'):
+  userPresets=load_json(get('mapTokens', '{}'))
+  argsTokenload = args.last('loadtoken').lower()
+  for searchTokenload in userPresets:
+   if argsTokenload in searchTokenload.lower():
+    foundTokenload = userPresets.get(searchTokenload)
+    desc.append(f"Loaded token `{searchTokenload}`.")
+    break
+  out[targ.name].update({'token':foundTokenload})
+</drac2>
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:658]: new loadtoken feature

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 11]
```diff
@@ -583,6 +674,25 @@ if c:
...
+    err("You must separate the key and value with a colon when adding `-savetoken` entries.")
+  set_uvar("mapTokens",dump_json(userTokens).lower())
+  desc.append(f"Created the Token `{addNewToken.split(':')[0].upper()}`")
+ if args.last('deletetoken'):
+  userTokens = load_json(get('mapTokens','{}'))
+  argToken = args.last("deletetoken").lower()
+  userTokens.pop(argToken)
+  set_uvar("mapTokens",dump_json(userTokens).lower())
+  desc.append(f"Removed Token `{argToken.upper()}`")
+</drac2>
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:694]: new savetoken feature. adds it to users mapTokens uvar

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 12]
```diff
@@ -894,9 +1004,13 @@ if c:
...
-  return f"""{'' if args.last('silent') else f'-image "{finalMap}"'} -f "[Map]({finalMap})" """ + (f"""  -desc "{newline.join(desc)}"  """ if desc else "")
+ if not any(args.get(arg) for arg in ('?', 'help', 'spelllist', 'spellbook', 'maplist', 'bglist', 'backgroundlist', 'viewlist', 'objectlist', 'tokenlist')):
+  if args.last('silent'):
+   finalMap = f"""{map}{mapsize}{f"/{mapviewlocation}:{mapviewsize}" if mapviewsize and mapviewlocation else ""}/{f'_{"".join(walls)}/' if walls else ""}{f"@{mapoptions}/"if mapoptions else""}{f'{"/".join(objects)}/' if objects else ""}{'/'.join(people+overlays)}{f"?bg={mapbg}" if mapbg else ""}"""
+   return f"""-f "[Map]({finalMap})" """ + (f"""  -desc "{newline.join(desc)}"  """ if desc else "")
+  else:
+   finalMap = f"""{map}{mapsize}{f"/{mapviewlocation}:{mapviewsize}" if mapviewsize and mapviewlocation else ""}/{f'_{"".join(walls)}/' if walls else ""}{f"@{mapoptions}/"if mapoptions else""}{f'{"/".join(objects)}/' if objects else ""}{'/'.join(people+overlays)}{f"?bg={mapbg}" if mapbg else ""}"""
+   return f"""-image "{finalMap}" -f "[Map]({finalMap})" """ + (f"""  -desc "{newline.join(desc)}"  """ if desc else "")
 </drac2>
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:1014]: Updated the finalmap return for walls and objects

File: Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar [hunk 13]
```diff
@@ -933,85 +1047,93 @@ elif args.get('maplist') or args.get('bglist') or args.get('backgroundlist'):
...
+ # Add some pizazz
+ return objectPage + f"""-title "I've got a lovely bunch of objects"  """
+elif args.get('tokenlist'):
+     # Only grab the page selected, default to first
+ if tokenPagin:
+  tokenPage = f"""-desc "{newline.join(tokenPagin[ min(len(tokenPagin),args.last('page', 1, int)-1)] )}" """
+ else:
+  tokenPage = f"""-desc "No results found{f" for search term `{args.last('search')}`" if args.last('search') else ''}." """
+ # Add some pizazz
+ return tokenPage + f"""-title "Token for your thoughts"  """
```
Comment (nerdsanddragons) [inline Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar:1075]: objectlist and tokenlist output

**Review by nerdsanddragons — COMMENTED** [review]
Commented on what new features I added

**Comment by Croebh** [comment]
Manually merged in https://github.com/Croebh/Avrae-Customizations/commit/be093c91100a6af9ca693f1a2ff01592e52258a4

**Pull request closed without merging**